platform-sdk/swirlds-common/src/main/java/com/swirlds/common/crypto/engine/IntakeDispatcher.java

```diff
@@ -153,13 +146,13 @@ public void shutdown() {
      */
     private void execute() {
         while (running) {
-            try {
-                final List<Element> workItems = backingQueue.take();
-                if (!workItems.isEmpty()) {
-                    executorService.submit(handlerSupplier.apply(provider, workItems));
-                }
-            } catch (InterruptedException ex) {
-                Thread.currentThread().interrupt();
+            final List<Element> workItems = backingQueue.poll();
+            if (workItems == null) {
+                Thread.yield();
```

Comment:
`Thread.onSpinWait()` is preferable here.

Reply:
updated